src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -67,8 +67,14 @@ private void OnSearchTextBoxTextChanged(object sender, TextChangedEventArgs e)
             if (binding != null)
                 binding.UpdateSource();
 
-            if (ViewModel != null)
-                ViewModel.SearchCommand.Execute(null);
+            if (ViewModel != null) 
+            {
+                Dispatcher.BeginInvoke(new Action(() =>
+                {
+                    // SearchTextBox.Focus();
+                    ViewModel.SearchAutoCompleteCandidates(SearchTextBox.Text);
```

Comment:
Would like to know the consequence of not using ViewModel command here

Reply:
I'll let @reddyashish add more info - but my thought is that this means the search is a special search (as evidenced by the fact that it uses just a `string.contains` on the name of the node.

As a consequence it also means the search cannot be started from a ViewExtension.

Commands make a lot of sense for operations we want to automate via tests, or which should be apis.
Don't have a strong feeling on this one.

Reply:
I thought whether implementing a viewcommand does not relate to which search algorithm we are using. If we do, I expect us instead of using the existing command but implement a new one dedicated for our needs with search elements already replaced for search view model so that we are not going over every searchElements in the pool but we still encounter for any search key words defined. For example, I can still search for cube to get to cuboid, etc.

I have no strong feeling about view commands but would like to validate what we missing if not using the search algorithm.

Reply:
descriptions will not be searched, weights will not be used, keywords will not be used.
Theres probably more.

Reply:
As of now, just matching the name of the node. Do we want to match the description or any other node property?

Reply:
I will review tonight again. it does seem there's gap not covered by naming comparison

Reply:
Thinking about this more, I'm fine with this approach given we may not want to be limited to the existing search and unclear if user would like to search through description yet